doc/classes/CanvasItem.xml

```diff
@@ -647,17 +648,17 @@
 		</signal>
 		<signal name="hidden">
 			<description>
-				Emitted when becoming hidden.
+				Emitted when the [CanvasItem] becomes hidden, either because its own [member visible] property was set to [code]false[/code] or the [member visible] property of one of its [CanvasItem] ancestors was set to [code]false[/code].
 			</description>
 		</signal>
 		<signal name="item_rect_changed">
 			<description>
-				Emitted when the item's [Rect2] boundaries (position or size) have changed, or when an action is taking place that may have impacted these boundaries (e.g. changing [member Sprite2D.texture]).
+				Emitted when the [CanvasItem]'s boundaries (position or size) change, or when an action took place that may have affected these boundaries (e.g. changing [member Sprite2D.texture]).
 			</description>
 		</signal>
 		<signal name="visibility_changed">
 			<description>
-				Emitted when the visibility (hidden/visible) changes.
+				Emitted when the [CanvasItem]'s visibility changes, either because its own [member visible] property changed or because the visility of one of its [CanvasItem] ancestors changed (equivalent to a change in [method is_visible_in_tree]).
```

Comment:
This is formulated in a consfusing manner / can be easily misinterpreted. This signal is emitted when either `visible` changes, or the value returned by `is_visible_in_tree()` changes. As simple as that.

IMO _"when the [CanvasItem]'s visibility changes"_ adds unneeded confusion, as it's not exactly explained what _"visibility changes"_ means, it only lists what could be causing it.

It can be easily misinterpreted that e.g. if parent's `visible = false` then changing `self.visible` won't result in emission of this signal as it's said to be emitted when _"visibilty changes"_. But if `is_visible_in_tree()` keeps returning false then visibility doesn't change? Or does it? Dang, what's the mentioned _"visibilty"_?! :thinking:

So something like this makes more sense to me (not sure about exact wording):
```suggestion
				Emitted either when the [CanvasItem]'s own [member visible] property changes, or when [method is_visible_in_tree] would return different value (because the visility of one of its [CanvasItem] ancestors changed).
```

Reply:
Good suggestion. Will make a follow-up PR for this.

Reply:
follow up here: https://github.com/godotengine/godot/pull/97367